subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractConsoleFunctionalSpec.groovy

```diff
@@ -46,6 +46,24 @@ class AbstractConsoleFunctionalSpec extends AbstractIntegrationSpec {
         return styledString
     }
 
+    /**
+     * Grabs the grouped output of a task
+     * @param taskName a fully qualified task name e.g. :project:task1
+     * @return
+     */
+    protected String taskOutputFrom(String taskName) {
```

Comment:
That's sort of what I was thinking about with this comment: https://github.com/gradle/gradle/pull/2082#discussion_r117327145

We can implement it with fancy regexes, but I think we should have some sort of programmatic interface that represents the final rich output of a build.  In that case, we can either implicitly combine multiple groups together or allow them to be queried separately.